blog/2023-05-03-understanding-entropy-the-key-to-secure-cryptography-and-randomness/index.md

```diff
@@ -0,0 +1,141 @@
+
+- **Indexing:** Some database systems use random numbers as a way of generating unique keys or IDs for indexing purposes. For example, MongoDB uses a 12-byte ObjectID that includes a random value, a timestamp, and other information as a unique identifier for each document.
+
+- **Unique values:** Some database systems use random numbers as a way of generating unique values for primary keys or other constraints. For example, PostgreSQL provides the uuid-ossp extension, which includes functions for generating UUIDs using random numbers as a source of entropy.
+
+In summary, the use of entropy in database systems varies depending on the specific use case and implementation.
+
+### Kubernetes
+
+Kubernetes can use entropy in various ways, depending on the specific use case and configuration. Here are some examples:
+
+- **Encryption:** Kubernetes can encrypt sensitive data, such as secrets and configuration files, using encryption keys that rely on random numbers as a source of entropy. By default, Kubernetes uses AES-256 encryption for data at rest and TLS encryption for data in transit.
+
+- **Random number generation:** Kubernetes can generate random numbers for various purposes, such as generating unique IDs for resources or generating tokens for authentication and authorization. For example, Kubernetes uses random numbers to generate session tokens for user authentication in its dashboard.
+
+- **Load balancing:** Kubernetes can use random numbers as part of its load balancing algorithms, such as when distributing traffic across multiple replicas of a service.
+
+- **Scaling:** Kubernetes can use random numbers as part of its scaling algorithms, such as when selecting nodes for scaling or selecting replicas for scaling.
+
+In general, Kubernetes relies on a sufficient amount of high-quality entropy to ensure the security and reliability of its operations, especially for encryption and authentication.
+
+## Entropy in Cloud VMs
+
+Cloud-provided VMs typically have access to hardware-generated entropy, but the specific implementation depends on the cloud provider and the underlying infrastructure.
+
+Most cloud providers recognize the importance of hardware-generated entropy for their customers' cryptographic operations and security needs. They often utilize hardware random number generators (HRNGs) or hardware security modules (HSMs) in their infrastructure to provide a reliable source of entropy.
+
+In general, cloud providers expose this entropy to VMs through virtual devices or interfaces. This allows VMs to access the hardware-generated entropy as if it were a local resource.
+
+- **AWS (Amazon Web Services)**
```

Comment:
Same issue here as in the uptime blog, this list doesn't appear properly in HTML. Adding the ":" is the faster solution.

Reply:
Fixed.